runtime/src/main/java/org/corfudb/runtime/collections/UFOStoreBase.java

```diff
@@ -0,0 +1,145 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * UFOStore is a thin layer over CorfuStore that provides certain metadata management
```

Comment:
Maybe the comment is off or the name is off but should this be in corfu side ?

Reply:
this is the *only* way we can write tests in Corfu to catch regressions that would impact verticals. 
It is code duplication across repos but the ability to catch regressions in my opinion outweighs the duplication.